react/Summary.tsx

```diff
@@ -1,48 +1,73 @@
 import React from 'react'
 import { ExtensionPoint } from 'vtex.render-runtime'
 import { Button } from 'vtex.styleguide'
+import SummaryItem from './components/SummaryItem'
 
-import styles from './styles.css'
+const minTotalizerValue = 0
+
+const getTotalizerById = (id: string, totalizers: Totalizer[]) => {
+  return totalizers.filter(totalizer => {
+    return totalizer.id === id
+  })
+}
+
+const Summary: StorefrontFunctionComponent<SummaryProps> = ({
+  totalizers,
+  total,
+  currency,
+}) => {
+  const subtotal = getTotalizerById('Items', totalizers)
+  const delivery = getTotalizerById('Shipping', totalizers)
+  const tax = getTotalizerById('CustomTax', totalizers)
 
-const Summary: StorefrontFunctionComponent<SummaryProps> = () => {
   return (
-    <div className={`${styles.container} flex-column pv6 ph4`}>
-      <h3 className="mid-gray ph4">Summary</h3>
-      <br></br>
-      <div className="flex ph4 pv4"> 
-        <div className="fl w-60 pa2 b">Subtotal</div>
-        <div className="w-40 pa2 tr">$100000202</div>
-      </div>
-
-      <div className="b--black-20 bt ph4">
-        <p className="b">Delivery</p>
-      </div>
-
-      <div className="b--black-20 bt ph4">
-        <ExtensionPoint id="coupon"/>
-      </div>
-
-      <div className="b--black-20 bt flex ph4 pv4">
-        <div className="fl w-60 pa2 b">Tax</div>
-        <div className="w-40 pa2 tr">$100000202</div>
-      </div>
-
-      <div className="bg-near-white b--black-20 bt ph4 pv4">
-        <div className="flex">
-          <div className="fl w-60 pa2 b">Total</div>
-          <div className="w-40 pa2 tr">$100000202</div>
-        </div>
-
-        <div className="pv4">
-          <Button variation="primary" size="large" block>CHECKOUT</Button>
-        </div>
-      </div>
+    <div className="ph5 ph0-l">
+      <h5 className="t-heading-5 mt6 mb6 mt8-l pt8-l">Summary</h5>
+      <ExtensionPoint id="coupon" />
+
+      <SummaryItem
```

Comment:
You should not instanciate each `SummaryItem`. It's better if you iterate through the totalizers array and declare summary item once like this:

```
totalizers.map(totalizer => (
    <SummaryItem
        label="Delivery"
        value={totalizer && totalizer.value || minTotalizerValue}
        large={false}
        currency={currency}
    />
))

Reply:
done!